pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -411,6 +411,16 @@ func MetricsAggregateFor(pod *kube_core.Pod) ([]*mesh_proto.PrometheusAggregateM
 	return aggregateConfig, nil
 }
 
+func mergeLabels(existingLabels map[string]string, podLabels map[string]string) map[string]string {
```

Comment:
From the usage point of view I guess I'd expect input params to be unchanged as long as 
`mergeLabels` returns map. IMO we should either return `maps.Clone` or return nothing. Given the concurrency issues we sometimes have with maps I'd go with `maps.Clone`

Reply:
+1, we should avoid editing exisitng map, we might hit concurrency read/write

Reply:
updated